mountpoint-s3/src/prefetch.rs

```diff
@@ -318,55 +357,12 @@ where
             }
 
             let part_len = part_bytes.len() as u64;
-            let result = response.extend(part_bytes);
-            match result {
-                Ok(()) => {}
-                Err(e @ IntegrityError::ChecksumMismatch(_, _)) => {
-                    // cancel inflight tasks
-                    self.current_task = None;
```

Comment:
nice that we've cleaned this up, could've been misleading why we do the cleanup twice